### ISSUE TITLE
Consider ppf() option

### ISSUE THREAD

**Issue by cliffckerr** [starsimhub/starsim#405]
It's slower, but gives the most functionality since can draw the random numbers independently of everything else to do with the distribution:
```py
import numpy as np
import sciris as sc
import scipy.stats as sps

reps = 100
n = int(1e6)

rng = np.random.default_rng()

with sc.timer('numpy'):
    for rep in range(reps):
        rng.normal(5, 1, size=n)

with sc.timer('scipy'):
    for rep in range(reps):
        sps.norm(5, 1).rvs(n)

with sc.timer('scipy ppf'):
    for rep in range(reps):
        q = rng.random(n)
        sps.norm(5, 1).ppf(q)
        
# numpy: 0.959 s
# scipy: 1.41 s
# scipy ppf: 2.88 s
```

Too crazy to use numpy by default, then switch to scipy.ppf if we need UID-specific array or callable parameters?

**Comment by daniel-klein** [comment]
Thanks @cliffckerr! Each distribution is implemented differently. For example, I believe numpy distributions are implemented in distributions.c, see here for normal: https://github.com/numpy/numpy/blob/main/numpy/random/src/distributions/distributions.c#L137

Scipy has variable implementations, with some distributions implemented using clever algorithms and others just calling ppf directly.

Before making a decision here, I recommend scanning over each of the distributions we typically use to assess performance.

**Comment by daniel-klein** [comment]
Also might want to investigate the case where `q` is selected outside the `rep` loop.

**Comment by cliffckerr** [comment]
Closed by #420
